### ISSUE TITLE
Add an "implementers guide" for developing backends

### ISSUE THREAD

**Issue by freakboy3742** [beeware/toga#3026]
### What is the problem or limitation you are having?

As noted in #3022, there is very little documentation or formal interface for someone seeking to implement a new backend, or implement a missing piece of an existing backend.

### Describe the solution you'd like

Add an implementation guide to the docs. This could cover details such as:

* An overview of the overall structure
* A guide to "first light" for a new backend (what to implement, etc)
* A list of the pieces that need to be implemented (what is required, what is optional but recommended, what is fully optional)
* The formal interface for each piece of the code that needs to be implemented
* Any known quirks, or behavioural edge cases that implementers should look out for

### Describe alternatives you've considered

A formal ABC or Protocol specification. This was the original form proposed in #3022. The downside is that introducing this as formal code has a non-zero impact on runtime, *won't* capture all the underlying details (such as behavioural and semantic details), and is only of interest to a relatively small audience.

**Comment by johnzhou721** [comment]
Having written a Qt backend, and planning to do WPF and WinUI once I get top-level + Apple things straightened out, I'd like to work on this.

Doing this in parallel with other work since this is docs / writing while others is mostly technical.

I'll open a draft PR with only stub changes, and then start writing on there.

**Comment by kattni** [comment]
Hi, @johnzhou721. As you know, the most of the core team is on leave for the holidays. We'll get back to you within a week with any necessary followup. Thanks!

**Comment by freakboy3742** [comment]
I'd definitely encourage you to write up anything you know/found in the process of setting up the Qt backend. 

Regarding future work - I honestly wouldn't worry too much about WPF at this point. Even Microsoft pretty much considers it a dead end, AFAICT. WinUI3 would be worthwhile, though - if only because it appears to have a richer set of base widgets.
